Source/Editor/DualityEditor/Forms/ObjectRefSelectionDialog.cs

```diff
@@ -0,0 +1,281 @@
+
+				foreach (Component currentComponent in Scene.Current.FindComponents(this.FilteredType))
+				{
+					ReferenceNode tmpNode = new ReferenceNode(currentComponent);
+
+					this.Model.Nodes.Add(tmpNode);
+				}
+			}
+			else
+			{
+				Type tmpFilteredType = typeof(Resource);
+				if (this.FilteredType.IsGenericType)
+				{
+					tmpFilteredType = this.FilteredType.GetGenericArguments()[0];
+					this.Text = string.Format("Select a {0} Resource", tmpFilteredType.GetTypeCSCodeName(true));
+				}
+				else
+				{
+					this.Text = "Select a Resource";
+				}
+
+				foreach (IContentRef contentRef in ContentProvider.GetAvailableContent(tmpFilteredType))
+				{
+					ReferenceNode tmpNode = new ReferenceNode(contentRef);
+
+					this.Model.Nodes.Add(tmpNode);
+				}
+			}
+
+			this.objectReferenceListing.NodeFilter += this.NodeFilter;
+
+			foreach (var treeNodeAdv in this.objectReferenceListing.AllNodes)
+			{
+				ReferenceNode tmpNode = treeNodeAdv.Tag as ReferenceNode;
+
+				treeNodeAdv.IsExpanded = true;
+
+				if (tmpNode != null && tmpNode.Path == this.ResourcePath)
+				{
+					this.objectReferenceListing.SelectedNode = treeNodeAdv;
+				}
+			}
+
+			this.objectReferenceListing.EndUpdate();
+			this.txtFilterInput.Focus();
+		}
+
+		private void ResourceListingOnDoubleClick(object sender, EventArgs eventArgs)
+		{
+			if (this.objectReferenceListing.SelectedNode == null)
+			{
+				this.ResourceReference = null;
+				this.GameObjectReference = null;
+				this.ComponentReference = null;
+
+				return;
+			}
+
+			this.ResourceListingOnSelectedIndexChanged(sender, eventArgs);
+			this.AcceptButton.PerformClick();
+		}
+
+		private void ResourceListingOnSelectedIndexChanged(object sender, EventArgs eventArgs)
+		{
+			if (this.objectReferenceListing.SelectedNode == null)
+			{
+				this.ResourceReference = null;
+				this.GameObjectReference = null;
+				this.ComponentReference = null;
+
+				return;
+			}
+
+			ReferenceNode node = this.objectReferenceListing.SelectedNode.Tag as ReferenceNode;
+
+			if (node == null)
+			{
+				return;
+			}
+
+			this.ResourceReference = node.ResourceReference;
+			this.GameObjectReference = node.GameObjectReference;
+			this.ComponentReference = node.ComponentReference;
+		}
+
+		private void txtFilterInput_TextChanged(object sender, EventArgs e)
+		{
+			this.objectReferenceListing.UpdateNodeFilter();
+
+			this.objectReferenceListing.SelectedNode = this.objectReferenceListing.AllNodes
+				.Where((node, index) => { return node.IsHidden == false; }).First();
+		}
+
+		private bool NodeFilter(TreeNodeAdv nodeAdv)
+		{
+			ReferenceNode tmpNode = nodeAdv.Tag as ReferenceNode;
+			string tmpFilterValue = this.txtFilterInput.Text.ToLowerInvariant();
+
+			return tmpNode != null &&
+			    ( 
+					tmpNode.Name.ToLowerInvariant().Contains(tmpFilterValue) ||
+					tmpNode.Path.ToLowerInvariant().Contains(tmpFilterValue)
+				);
+		}

```

Comment:
Minor thing but formatting seems to be a bit off

Reply:
Replaced the spaces with tabs.